Testament now checks OS, CPU, endianess, bitsizes 8-64 in discard statements

quantimnot opened the pull request:
I'm not sure what the correct way to disable specific platforms was before, but this PR will allow the `discard` statement work for any of the OS/CPU values in `compiler/platform`.

quantimnot committed:
Testament checks host/cpu in compiler/platform for discard spec statement

Araq commented on testament/specs.nim:
```diff
@@ -364,7 +366,18 @@ proc parseSpec*(filename: string): TSpec =
+          block checkHost:
+            for os in platform.OS:
+              if value == os.name.normalize:
+                if value == hostOS.normalize:
```
Rewrite the logic or add a comment, so in the end hostOS should be the same as os.name? Why?

quantimnot replied:
See new code comments

quantimnot committed:
Remove duplicated value checks, add comments, add all cpu bit sizes and endianesses

quantimnot commented:
In my last commit, I removed case branches that would be covered by the host OS/CPU check in the else branch. I also added all bit size branches for 8, 16, and 64. I included both the define names and the form '(8|16|32|64)bit' for the bitsize. I also added branches for little and big endianess.
If this is overreaching or something, let me know and I'll revert it.
